src/account-abstraction.md

```diff
@@ -0,0 +1,168 @@
+
+
+
+The ```__execute__``` function is the only one that receives a reference to the contract state because it’s the only one likely to either modify its internal state or to modify the state of another smart contract and thus to require the payment of gas fees for its execution. The other two functions, ```__validate__``` and ```is_valid_signature```, are read-only and shouldn’t require the payment of gas fees. For this reason they are both receiving a snapshot of the contract state instead.
+
+
+<h3> Executing Transactions </h3>
+
+The ```__execute__``` function is responsible for executing the transaction. In this minimal account contract we will only execute a single call. 
+```
+fn __execute__(self: @ContractState, calls: Array<Call>) -> Array<Span<felt252>> {
+            let Call{to, selector, calldata } = calls.at(0);
+            let _res = starknet::call_contract_syscall(*to, *selector, calldata.span()).unwrap();
+            let mut res = ArrayTrait::new();
+            res.append(_res);
+            res
+        }
+```
+
+The ```__execute__``` function calls the call_contract_syscall function from the starknet module. This function executes the call and returns the result. The ```call_contract_syscall``` function is a Starknet syscall, which means that it is executed by the Starknet protocol. The Starknet protocol is responsible for executing the call and returning the result. The Starknet protocol will also validate the call, so we do not need to validate the call in the ```__execute__``` function.
+
+<h3> Validating Transactions </h3>
+
+The ```is__valid__signature``` function is responsible for the validation, returning ```VALIDATED``` if the signature is valid. The ```VALIDATED``` constant is imported from the starknet module.
+ 
+
+```
+use starknet::VALIDATED;
+```
+
+Notice that the ```is_valid_signature``` function accepts all the transactions as valid. We are not storing a public key in the contract, so we cannot validate the signature.
+
+
+ 
+```
+fn is_valid_signature(
+            self: @ContractState, hash: felt252, signature: Array<felt252>
+        ) -> felt252 {
+            // No signature is required so any signature is valid.
+            VALIDATED
+        }
+```
+
+```is__valid__signature```is designed to validate a signature for a given hash within the context of a smart contract's state. This function takes three parameters: a reference to the contract's state `(self: @ContractState)`, the hash of the data that purportedly needs to be signed `(hash: felt252)`, and the actual signature represented as an array of felt252 values `(signature: Array<felt252)`. Instead of performing actual cryptographic signature validation, the function simplistically returns a constant `VALIDATED`, indicating that any signature provided is considered valid.
```

Comment:
Use simple english.
Use bullet list or any other formats for better readability.